app/utilities/metadata_parser.py

```diff
@@ -146,6 +146,30 @@ def convert_schema_name(self, data, **kwargs):
             )
         return data
 
+    @post_load
+    def update_response_id(
+        self, data, **kwargs
+    ):  # pylint: disable=no-self-use, unused-argument
+        """
+        If response_id is present : return as it is
+        If response_id is not present : Build response_id from ru_ref,collection_exercise_sid,eq_id and form_type
+                                        and updates metadata with response_id
+
+
+        """
+        if data.get("response_id"):
+            return data
+        eq_id = data.get("eq_id")
+        form_type = data.get("form_type")
+        ru_ref = data.get("ru_ref")
+        collection_exercise_sid = data.get("collection_exercise_sid")
```

Comment:
Minor, and non-blocker but these could be bracket access since they are mandatory.
```py
ru_ref = data.["ru_ref"[
collection_exercise_sid = data["collection_exercise_sid"]]
```